frontend/packages/dev-console/src/components/monitoring/MonitoringPage.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { Firehose, HorizontalNav, PageHeading, history } from '@console/internal/components/utils';
+import { ALL_NAMESPACES_KEY } from '@console/internal/const';
+import { ProjectModel } from '@console/internal/models';
+import { TechPreviewBadge } from '@console/shared';
+import NamespacedPage, { NamespacedPageVariants } from '../NamespacedPage';
+import ProjectListPage from '../projects/ProjectListPage';
+import MonitoringDashboard from './MonitoringDashboard';
+import MonitoringMetrics from './MonitoringMetrics';
+import { connectActiveNamespace, ConnectActiveNamespaceProps, redirectURI } from './utils';
+
+const handleNamespaceChange = (newNamespace: string): void => {
+  history.push(redirectURI(newNamespace));
+};
+
+export const MonitoringPage: React.FC<ConnectActiveNamespaceProps> = ({
+  activeNamespace,
+  ...props
+}) => {
+  const allNamespaces = activeNamespace === ALL_NAMESPACES_KEY;
+
+  return (
+    <>
+      <Helmet>
+        <title>Monitoring</title>
+      </Helmet>
+      <NamespacedPage
+        hideApplications
+        variant={NamespacedPageVariants.light}
+        onNamespaceChange={handleNamespaceChange}
+      >
+        {allNamespaces ? (
+          <ProjectListPage badge={<TechPreviewBadge />} title="Monitoring">
+            Select a project to view monitoring metrics
+          </ProjectListPage>
+        ) : (
+          <Firehose
+            resources={[
+              {
+                kind: ProjectModel.kind,
+                kindObj: ProjectModel,
+                name: activeNamespace,
+                namespace: activeNamespace,
+                isList: false,
+                prop: 'obj',
+              },
+            ]}
+          >
+            <PageHeading badge={<TechPreviewBadge />} title="Monitoring" />
+            <HorizontalNav
+              pages={[
+                {
+                  href: '',
+                  name: 'Dashboard',
+                  component: MonitoringDashboard,
+                },
+                {
+                  href: 'metrics',
+                  name: 'Metrics',
+                  component: MonitoringMetrics,
+                },
+              ]}
+              className={`co-m-${ProjectModel.kind}`}
+              match={props.match}
+              noStatusBox
+            />
+          </Firehose>
+        )}
+      </NamespacedPage>
+    </>
+  );
+};
+
+export default connectActiveNamespace(MonitoringPage);
```

Comment:
Root-level pages don't need to connect to the store to get the namespace. Each page, using the `type: 'Page/Route'` plugin, will be created using React-Router. This injects a `match` prop. Inside it is the namespace.

```
props.match.params
{ns: "<my-current-namespace>"}
```
The `ns` is the param you gave to the route URL in the plugin file `'/dev-monitoring/ns/:ns'`.

Checking for all-projects becomes a check against the missing param.

Reply:
@andrewballantyne removed connection to store & now using the `match` prop to get namespace